docs/hub/how-to-downstream.md

```diff
@@ -0,0 +1,53 @@
+# How to integrate downstream methods in your library
+
+In the tutorial, you learned some basic functions for integrating the Hub into your library. Functions that allow your library to download files from the Hub are referred to as *downstream* functions. This guide introduces additional downstream methods you can integrate in your library. You will learn how to:
+
+* Download a file without caching it on your disk.
+* Download a file and cache it on your disk.
+* Download all the files in a repository.
+
+## `hf_hub_url`
+
+Use `hf_hub_url` to retrieve the URL of a specific file to download by providing a `filename`.
+
+![/docs/assets/hub/repo.png](/docs/assets/hub/repo.png)
+
+```python
+>>> from huggingface_hub import hf_hub_url
+>>> hf_hub_url(repo_id="lysandre/arxiv-nlp", filename="config.json")
+'https://huggingface.co/lysandre/arxiv-nlp/resolve/main/config.json'
+```
+
+Specify a particular file version by providing the file revision. The file revision can be a branch, a tag, or a commit hash:
+
+```python
+>>> hf_hub_url(repo_id="lysandre/arxiv-nlp", filename="config.json", revision="877b84a8f93f2d619faa2a6e514a32beef88ab0a")
+'https://huggingface.co/lysandre/arxiv-nlp/resolve/877b84a8f93f2d619faa2a6e514a32beef88ab0a/config.json'
+```
+
+## `cached_download`
+
+The `cached_download` function is useful for downloading and caching a file on your local disk. Once it is stored in your cache, you don't have to redownload the file the next time you use it. This is a hands-free solution for staying up to date with new file versions. When one of your downloaded files is updated in the repository, it is automatically downloaded and stored for you.
+
+Begin by downloading your file with `hf_hub_url`, and then pass the specified URL to `cached_download`:
```

Comment:
The `hf_hub_url` doesn't download the file, it retrieves the link as you've highlighted above :)
```suggestion
Begin by retrieving your file's URL with `hf_hub_url`, and then pass the specified URL to `cached_download` to download it:
```